File: Chapter09/codes_chapter_9/python/add_documents.py

import pysolr

solr = pysolr.Solr('http://localhost:8983/solr/techproducts', timeout=5)

solr.add([
    {
        'id':'id1',
        'name':'product1'
    },
    {
        'id':'id2',
        'name':'product2'
    },
    {
        'id':'idn',
        'name':'productn'
    }
])

File: Chapter09/codes_chapter_9/ruby/add_documents.rb
require 'rsolr'

solr = RSolr.connect :url => 'http://localhost:8983/solr/techproducts'

response = solr.add([
	{ :id => 'id1', :name => "product1"},
	{ :id => 'id2', :name => "product2"},
	{ :id => 'idn', :name => "productn"}
	])

solr.commit

puts response
